### PR TITLE
Fixed issue #18471: Data Entry: Data entry with multiple languages does not work

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/helpers/common_helper.php [hunk 1]
```diff
@@ -2335,9 +2335,8 @@ function languageDropdown($surveyid, $selected)
     $baselang = Survey::model()->findByPk($surveyid)->language;
     array_unshift($slangs, $baselang);
     $html = "<select class='listboxquestions' name='langselect' onchange=\"window.open(this.options[this.selectedIndex].value, '_top')\">\n";
-
     foreach ($slangs as $lang) {
-        $link = Yii::app()->homeUrl . ("/admin/dataentry/sa/view/surveyid/" . $surveyid . "/lang/" . $lang);
+        $link = Yii::app()->createUrl("admin/dataentry/sa/view/surveyid/". $surveyid . "/lang/" . $lang);
```
Comment (Shnoulle) [inline application/helpers/common_helper.php:2339]: I personnaly prefer `Yii::app()->createUrl("admin/dataentry/sa/view/surveyid/". $surveyid, ['lang' => $lang])` for me lang is a parameter, i know sureyid too, but less clear

**Review by Shnoulle — APPROVED** [review]
OK
Needed :)

File: application/helpers/common_helper.php (hunk 1 above)
Comment (gabrieljenik) [reply application/helpers/common_helper.php:2339]: These are written with different patterns all over the codebase.... We should set a std

Comment (Shnoulle) [reply application/helpers/common_helper.php:2339]: Yes , i know …

Comment (olleharstedt) [reply application/helpers/common_helper.php:2339]: Something for the code quality guide?

Comment (gabrieljenik) [reply application/helpers/common_helper.php:2339]: Defintely, I will let you set the standard :)

**Pull request merged**